### PR TITLE
Remove next_in_quadrant

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Another field that doesn't need to be located in the Entity base. We rebuild the spatial index on load so this field was never read from the SV6/4 file.

### CODE REVIEW

File: src/openrct2/world/Sprite.cpp [hunk 1]
```diff
@@ -32,6 +33,7 @@ static rct_sprite _spriteList[MAX_SPRITES];
 static bool _spriteFlashingList[MAX_SPRITES];
 
 uint16_t gSpriteSpatialIndex[SPATIAL_INDEX_SIZE];
+std::array<std::vector<uint16_t>, SPATIAL_INDEX_SIZE> gSpriteSpatialIndex2;
```
Comment (duncanspumpkin) [inline src/openrct2/world/Sprite.cpp:36]: can delete the 2 and remove the old spatial index.

File: src/openrct2/world/Sprite.h [hunk 2]
```diff
@@ -332,25 +332,75 @@ template<typename T, uint16_t SpriteBase::*NextList> class EntityIterator
     using iterator_category = std::forward_iterator_tag;
 };
 
+template<typename T> class EntityTileIterator
+{
+private:
+    const std::vector<uint16_t>& vec;
```
Comment (duncanspumpkin) [inline src/openrct2/world/Sprite.h:338]: only need the end iterator so might as well take two const iterators

**Comment by duncanspumpkin** [comment]
Could do with doing some performance checking just to make sure it doesn't change anything.

**Comment by duncanspumpkin** [comment]
Note still need to rebuild the next_in_quadrant value for exporting.

File: src/openrct2/rct2/S6Exporter.cpp [hunk 3]
```diff
@@ -990,7 +990,6 @@ void S6Exporter::ExportSprite(RCT2Sprite* dst, const rct_sprite* src)
 void S6Exporter::ExportSpriteCommonProperties(RCT12SpriteBase* dst, const SpriteBase* src)
 {
     dst->sprite_identifier = src->sprite_identifier;
-    dst->next_in_quadrant = src->next_in_quadrant;
```
Comment (duncanspumpkin) [inline src/openrct2/rct2/S6Exporter.cpp:993]: still needed

**Comment by IntelOrca** [comment]
I notice your iterators are enumerating `Peep*`, should they be enumerating `Peep&` instead? It shouldn't ever return a null peep.

I think it would be good to make everything to do with the spatial index a class of its own. But that could be done in another PR.

**Pull request merged**